waterbutler/providers/rushfiles/provider.py

```diff
@@ -131,4 +177,53 @@ async def zip(self, path: WaterButlerPath, **kwargs) -> asyncio.StreamReader:
         # I will check and if there is, it may be more efficient then default behaviour.
         return super().zip(path, kwargs)
 
-    
+    async def _folder_metadata(self,
+                               path: RushFilesPath,
+                               raw: bool=False) -> List[Union[BaseRushFilesMetadata, dict]]:
+        share_id = self.share['id']
+        inter_id =  path.identifier
+
+        response = await self.make_request(
+            'GET',
+            self.build_url(str(share_id), 'virtualfiles', inter_id, 'children'),
+            expects=(200, 404,),
+            throws=exceptions.MetadataError,
+        )
+
+        if response.status == 404:
+            raise exceptions.NotFoundError(path)
+        res = await response.json()
+
+        if raw:
+            return res['Data']
+        else:
+            ret = []
+            for data in res['Data']:
+                ret.append(RushFilesFileMetadata(data, path))
```

Comment:
Folder may contain also other folders. Not only files. You need to check for it and create RushFilesFolderMetadata when necessary.